firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -124,26 +129,25 @@ public static FirebaseAppDistribution getInstance(@NonNull FirebaseApp app) {
   @NonNull
   public Task<AppDistributionRelease> updateToLatestRelease() {
 
-    TaskCompletionSource<AppDistributionRelease> taskCompletionSource =
-        new TaskCompletionSource<>();
+    if (updateToLatestReleaseTaskCompletionSource != null
+        && !updateToLatestReleaseTaskCompletionSource.getTask().isComplete()) {
+      updateToLatestReleaseCancellationSource.cancel();
+    }
+
+    updateToLatestReleaseCancellationSource = new CancellationTokenSource();
+    updateToLatestReleaseTaskCompletionSource =
+        new TaskCompletionSource<>(updateToLatestReleaseCancellationSource.getToken());
 
+    // todo: when persistence is implemented and user already signed in, skip signInTester
     signInTester()
-        .addOnSuccessListener(
-            new OnSuccessListener<Void>() {
-              @Override
-              public void onSuccess(Void unused) {
-                taskCompletionSource.setResult(null);
-              }
-            })
```

Comment:
I think there might be a lot of method nesting here. I feel we should move up the checkForUpdate call in here and maybe add the alert dialog in a different method.